### PR TITLE
#453 Fix 401 error

### PR DESCRIPTION
**Pull request by oubchid**

### CODE REVIEW

File: src/esn.inbox.libs/app/components/message-body/html/message-body-html.spec.js [hunk 1]
```diff
@@ -9,6 +9,15 @@ describe('The inboxMessageBodyHtml component', function() {
   var $compile, $rootScope, $timeout, newComposerService;
   var element;
 
+  const tokenAPIMock = {
+    getNewToken: function() {
+      return $q.when({ data: { token: '123' } });
```
Comment (tanandy) [inline src/esn.inbox.libs/app/components/message-body/html/message-body-html.spec.js:14]: what is 123 ?

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js [hunk 2]
```diff
@@ -49,23 +55,20 @@ describe('The jmapClientProvider service', function() {
...
+    jmapClientProvider.get()
+      .then(function(client) {
+        expect(client).to.be.an.instanceof(jmapDraft.Client);
+        expect(client.authToken).to.equal('Bearer jwt');
+        expect(client.apiUrl).to.equal('expected jmap api');
+        expect(client.downloadUrl).to.equal('expected jmap downloadUrl');
 
-      done();
-    }, done.bind(null, 'should resolve'));
+        done();
```
Comment (tanandy) [inline src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js:70]: do we have spaces here ?

File: src/esn.inbox.libs/app/components/message-body/html/message-body-html.spec.js (hunk 1 above)
Comment (oubchid) [reply src/esn.inbox.libs/app/components/message-body/html/message-body-html.spec.js:14]: It doesn't matter the value here, just a token value

Comment (rezk2ll) [reply src/esn.inbox.libs/app/components/message-body/html/message-body-html.spec.js]: this is a unit @tanandy

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js (hunk 2 above)
Comment (tanandy) [reply src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js:67]: yes, no ?

**Review by rezk2ll — APPROVED** [review]

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js (hunk 2 above)
Comment (rezk2ll) [reply src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js:67]: to match the indentation level, the `done` should be at the same level of the `expect`

image

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.js [hunk 3]
```diff
@@ -17,14 +16,14 @@ angular.module('esn.inbox.libs')
 
     function _initializeJmapClient() {
       return $q.all([
-        generateJwtToken(),
+        tokenAPI.getWebToken(true),
```
Comment (rezk2ll) [inline src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.js:19]:
```suggestion
        tokenAPI.getWebToken(),
```

we don't want a new token each time. we need to use the cached one

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.js [hunk 4]
```diff
@@ -17,14 +16,14 @@ angular.module('esn.inbox.libs')
...
         inboxConfig('api'),
         inboxConfig('downloadUrl')
-      ]).then(function(data) {
+      ]).then(function([{ data: jwt }, api, downloadUrl]) {
```
Comment (rezk2ll) [inline src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.js:22]:
```suggestion
      ]).then(function([{ data: jwt }, apiUrl, downloadUrl]) {
```

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.js [hunk 5]
```diff
@@ -17,14 +16,14 @@ angular.module('esn.inbox.libs')
...
         jmapClient = new jmapDraft.Client(dollarHttpTransport, dollarQPromiseProvider)
-          .withAPIUrl(data[1])
-          .withDownloadUrl(data[2])
-          .withAuthenticationToken('Bearer ' + data[0])
+          .withAPIUrl(api)
```
Comment (rezk2ll) [inline src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.js:24]:
```suggestion
          .withAPIUrl(apiUrl)
```

**Review by rezk2ll — CHANGES_REQUESTED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

**Review by alagane — APPROVED** [review]

**Pull request merged**